src/main/java/core/Parser.java

```diff
@@ -0,0 +1,212 @@
+package core;
+
+import exceptions.DukeException;
+
+/**
+ * Parses user input into usable terms for core.Duke.
+ */
+public class Parser {
+    private String userInput;
+    private final String keyword;
+
+    /**
+     * For enum attempt
+     */
+    public enum Keyword {
+        TODO, DEADLINE, EVENT, FIND
+    }
+
+    /**
+     * Constructor for Parser class.
+     *
+     * @param userInput The command to parse
+     */
+    public Parser(String userInput) {
+
+        this.userInput = userInput.trim();
+        String[] split = userInput.split(" ", 2);
+        if (split.length > 1) {
+            this.userInput = split[1];
+        } else {
+            this.userInput = "";
+        }
+        this.keyword = split[0].toLowerCase().trim();
+    }
+
+    /**
+     * Returns first word of user input.
+     * For example 'mark 1', 'mark' would be considered as the keyword.
+     *
+     * @return The command which the user has typed.
+     */
+    public String getKeyword() {
+        return keyword;
+    }
+
+    /**
+     * Returns integer from user input.
+     * This assumes that the second item in the user input is an integer.
+     * Example: 'mark 1', '1' is expected to be an integer.
+     *
+     * @return The integer value which the user has passed in.
+     * @throws DukeException Thrown when user input is invalid or missing.
+     */
+    public int extractIndexParams() throws DukeException {
+        String[] userInSplit = this.userInput.split(" ", 2);
+        if (userInSplit.length < 1) {
+            throw new exceptions.missing.Parameter(this.keyword);
+        }
+        try {
+            int ind = Integer.parseInt(userInSplit[0]);
+            return ind - 1; // Count starting from 0
```

Comment:
might be clearer if name the ind variable

Reply:
Thank you for your kind words.

Thank you for pointing out my mistake. It looks like I have missed out this particular part by accident.